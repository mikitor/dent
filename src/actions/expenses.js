import uuid from 'uuid';

const addExpense = ({ description = '', note = '', amount = 0, createdAt } = {}) => ({
  type: 'ADD_EXPENSE',
  expense: {
    id: uuid(),
    description,
    note,
    amount,
    createdAt
  }
});

const deleteExpense = ({ id } = {}) => ({
  type: 'DELETE_EXPENSE',
  id
});

const editExpense = (id, update) => ({
  type: 'EDIT_EXPENSE',
  id,
  update
});

export { addExpense, deleteExpense, editExpense };
