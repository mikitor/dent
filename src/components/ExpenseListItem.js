import React from 'react';
import moment from 'moment';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({ description, amount, createdAt, id }) => (
  <div>
    <Link to={`/edit/${id}`}>{description}</Link>
    <p>Amount: ${amount / 100}</p>
    <p>CreatedAt: {moment(createdAt).format('LL')}</p>
  </div>
);

export default ExpenseListItem;
