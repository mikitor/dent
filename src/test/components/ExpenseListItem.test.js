import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';
import { expenseOne } from '../fixtures/expenses';

it('should render ExpenseListItem component with props', () => {
  const wrapper = shallow(<ExpenseListItem {...expenseOne} />);
  expect(wrapper).toMatchSnapshot();
});
