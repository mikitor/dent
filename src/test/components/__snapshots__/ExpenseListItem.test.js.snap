// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem component with props 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/124"
  >
    Test one desc
  </Link>
  <p>
    Amount: $
    4
  </p>
  <p>
    CreatedAt: 
    January 1, 1970
  </p>
</div>
`;
